plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java

```diff
@@ -819,8 +819,12 @@ public String toString() {
             diskBuilder.append(" type='" + _diskType + "'");
             diskBuilder.append(">\n");
             if(qemuDriver) {
-                diskBuilder.append("<driver name='qemu'" + " type='" + _diskFmtType
-                        + "' cache='" + _diskCacheMode + "' ");
+                if (_deviceType == DeviceType.CDROM && _sourcePath == null) {
```

Comment:
What about extracting this bit of code that appends the disk information to a XML to a method called? Then, it is possible to further document it (when the cache mode is used and why it is not used sometimes), and of course it would enable us to unit test this logic of creating the disk entry.

Reply:
Please see my comments, this pr may be closed. The issue seems regression in qemu-kvm

Reply:
Ah, I am sorry. I came directly to the code.